### ISSUE TITLE
Potential bug in when CustomPortsNode updates it's input ports

### ISSUE THREAD

**Issue by cbaggers** [alelievr/NodeGraphProcessor#68]
If I have understood the point of this node correctly it should always have one more port than the max number of input edges is has had previously. However when I add an input edge it doesn't update the port UI until some other action forces it (for example adding a new output edge).

This clip shows the issue:

![multiNodeUpdateBug](https://user-images.githubusercontent.com/1579326/87981636-de05fa80-cad5-11ea-890e-4ebff28583da.gif)

In my code I worked around this by adding a custom view as follows:
```
[NodeCustomEditor(typeof(SomeNode))]
public class SomeNodeView : RealtimeNodeView
{
    public override void Enable()
    {
	nodeTarget.onAfterEdgeConnected += OnStructuralChange;
	nodeTarget.onAfterEdgeDisconnected += OnStructuralChange;
    }

    void OnStructuralChange(SerializableEdge _) => ForceUpdatePorts();
}
```

Is this intended behavior? Apologies if I missed something obvious.

**Comment by alelievr** [comment]
Hello,

There is indeed an issue here, this node should always have one more port to connect the input. I just check and it appears that the issue is fixed in the [port-propagation](https://github.com/alelievr/NodeGraphProcessor/tree/port-propagation) branch which should be merged soon (just need to finish some testing).

Thanks for the detailed bug report :)

**Comment by cbaggers** [comment]
Fantastic! Thanks a lot. I'll leave this ticket open for now and will test out the changes when they land.

**Comment by alelievr** [comment]
The branch have been merged in master at 4d689737ab2403c912f9339a42aefa8571093987
